packages/node-cli/src/core/CommandActionExecutor.js

```diff
@@ -83,20 +83,32 @@ module.exports = class CommandActionExecutor {
 				projectConfiguration: projectConfiguration,
 			});
 
-			this._commandOutputHandler.showSuccessResult(actionResult, command.formatOutputFunc);
+			if (!(actionResult instanceof ActionResult)) {
+				throw 'INTERNAL ERROR: Command must return an ActionResult object.';
+			}
 
-			if (actionResult && actionResult.operationResult) {
-				const operationResult = actionResult.operationResult;
-				if (operationResult.status === OperationResultStatus.SUCCESS && commandUserExtension.onCompleted) {
-					commandUserExtension.onCompleted(actionResult);
-				} else if (operationResult.status === OperationResultStatus.ERROR && commandUserExtension.onError) {
-					const error = SDKOperationResultUtils.getResultMessage(operationResult)
-						+ NodeUtils.lineBreak
-						+ SDKOperationResultUtils.getErrorMessagesString(operationResult);
-					commandUserExtension.onError(error);
+			if (actionResult.status === ActionResult.ERROR) {
+				if (actionResult.errorMessages && Array.isArray(actionResult.errorMessages) && actionResult.size > 0) {
+					throw actionResult.errorMessages;
+				}
+				else if (actionResult.resultMessage) {
+					throw actionResult.resultMessage;
+				}
+				else {
+					throw 'Unexpected Error' //TODO Code Review and Change message
```

Comment:
now that we always build the actionResult in the same class, we could homogenize this and have always the errors in actionResult.errorMessages, no? so that we only need to do
```
if (actionResult.status === ActionResult.ERROR) {
  throw actionResult.errorMessages;
}
```

Reply:
There is at least one command that have the errorMessage empty and the errorMessage is in resultMessage.

This is a problem in JavaCLI not in Nodejs... I don't want to touch JavaCLI since the story is already too big so I would let it as it is

Reply:
(and I have only found one case... I haven't tried edge cases) I will put a //TODO. I know this is not the best way of course, but we should study all JavaCLI cases

Reply:
ok just forget about it... there is a comment a David that resolves this.

I close the conversation